app/stac_api/pgtriggers.py

```diff
@@ -61,79 +80,114 @@ def generate_child_triggers(parent_name, child_name):
     ]
 
 
-def generates_asset_triggers():
-    '''Generates Asset triggers
+def asset_counter_trigger(count_table, value_field):
+    '''Creates triggers to adjust the asset summary on a counter table.
 
-    Those triggers act on `insert`, `update` and `delete` Asset event and do the followings:
-      - Update the `updated` and `etag` fields of the assets and their parents
-      - Update the parent collection summaries.
-      - Update the parent item `update_interval` by using the minimal aggregation of all of its
-        assets
+    Args:
+         count_table: tha table name to be updated (without prefix stac_api_)
+         value_field: summary field name on the asset
 
-    Returns: tuple
-        tuple for all needed triggers
+    Returns:
+        List of triggers
     '''
 
-    class UpdateCollectionSummariesTrigger(pgtrigger.Trigger):
+    class UpdateDeleteCounterTrigger(pgtrigger.Trigger):
```

Comment:
In all honesty, the naming of these classes confuse me a bit (and yes I know, you didn't start it).
I think it is supposed to say "update the delete-counter-trigger", right? Yet classes usually don't contain verbs I would argue? And it's especially confusing since "Update" is a common word in SQL :D

Reply:
My initial idea in this case was that the trigger is used in SQL Update and SQL Delete command, But I see how this is confusing. I refactored the naming of the functions and classes used in the triggers to try and make it more clear. Now there are functions that create "generic triggers" and functions that create triggers for a specific model (these start with `generates_`).